sdk/azidentity/example_test.go

```diff
@@ -64,7 +64,9 @@ func ExampleNewOnBehalfOfCredentialWithCertificate() {
 		// TODO: handle error
 	}
 
-	cred, err = azidentity.NewClientCertificateCredential(tenantID, clientID, certs, key, nil)
+	// userAssertion is the user's access token for the application. Typically it comes from a client request.
+	userAssertion := "TODO"
+	cred, err = azidentity.NewOnBehalfOfCredentialWithCertificate(tenantID, clientID, userAssertion, certs, key, nil)
```

Comment:
Can the `func` argument legitimately be `nil`, or is this a placeholder for the example? If it's the former, would be good to include that in the doc comment. If the latter, maybe add a comment here explaining it needs to be populated.

Reply:
This example is intended to show certificate auth but was calling the wrong constructor for that. This change is just me fixing it, no relation to the new feature. But to your question, there's no scenario for a nil `getAssertion` func because all token requests must contain a client assertion. As I have it now, passing `nil` will provoke a panic when the credential tries to authenticate; I'll make that a constructor error instead

Reply:
Derp, I misread that. Good idea on the `nil` check though given that the panic will manifest later, making it hard to diagnose.